src/Bicep.LangServer/BicepCompilationManager.cs

```diff
@@ -42,11 +45,19 @@ public BicepCompilationManager(ILanguageServerFacade server, ICompilationProvide
             this.scheduler = scheduler;
         }
 
-        public void RefreshCompilation(DocumentUri documentUri)
+        public void RefreshCompilation(DocumentUri documentUri, bool reloadBicepConfig = false)
         {
             var compilationContext = this.GetCompilation(documentUri);
+
             if (compilationContext is null)
             {
+                if (reloadBicepConfig &&
+                    workspace.TryGetSourceFile(documentUri.ToUri(), out ISourceFile? sourceFile) &&
+                    sourceFile is BicepFile)
+                {
+                    UpsertCompilationInternal(documentUri, null, sourceFile, reloadBicepConfig);
+                }
```

Comment:
I'd like @majastrz to take a look at the changes to this class - as the mechanism he added for the module registry work may be useful here for triggering changes on a source file with no syntax differences.

Reply:
This was the mechanism. Although, I'm planning to change the refresh part to take a module reference and have logic to compute which document URIs need to updated and then recompile them in optimal order.